App inifile generation testing document updates (#67)
Centiq/sap-hana: documentation/ansible/system-design-deployment/app/prepare-ini.md

1. `/sapmnt/<SID>/global`
1. `/sapmnt/<SID>/profile`

> ### Mounting SAP FileSystems on APP VMs

Reviewer: Strictly, the SCS is also an app VM, so this may be confusing. Perhaps `...on PAS and AAS VMs`?

1. `/sapmnt/<SID>/global`
1. `/sapmnt/<SID>/profile`

> ### Mounting SAP FileSystems on APP VM

Reviewer 2: ```suggestion
### Mounting SAP FileSystems on App VMs
```
Or do you only do it on one?

Reviewer 3: Needs doing once per Application Instance VM. Line 171 also directly references the PAS.

1. On the PAS VM as `root` ensure the mount points exist:

   `mkdir -p /usr/sap/{downloads,install/config,<SID>/SYS} /tmp/app_template /sapmnt/<SID>/{global,profile}`

1. Ensure the exported directories are mounted:
   1. `mount <scs-vm-IP>:/usr/sap/install /usr/sap/install`
   1. `mount <scs-vm-IP>:/usr/sap/<SID>/SYS /usr/sap/<SID>/SYS`
   1. `mount <scs-vm-IP>:/tmp/app_template /tmp/app_template`
   1. `mount <scs-vm-IP>:/sapmnt/<SID>/global /sapmnt/<SID>/global`
   1. `mount <scs-vm-IP>:/sapmnt/<SID>/profile /sapmnt/<SID>/profile`

### Database Content Load Prerequisites

The following requirements must be in place on the PAS DB VM before attempting the DB Content Load:

1. `<sid>adm` User must exist and must be a member of the `sapinst` group;
1. The user ID for `<sid>adm` must match the value provided to hdblcm (`2000` is used in this process);

> 1. The Directory `/sapmnt/S1D/global/` must be accessible to SWPM `chown <sid>adm:sapsys /sapmnt/<SID>/global`;

Reviewer 2: Should `S1D` in the initial path be `<sid>` to match the others, or the uppercase version of it `<SID>`?

1. `sapinst` group must exist.

### Generating unattended installation parameter `inifile` for Database Content Load

:hand: Both HANA and SCS instances must be installed, configured and online before completing the DB content load.

1. Make and change to a temporary directory:

   `sudo install -d -m 0777 <sid>adm -g sapinst "/tmp/db_workdir"; cd $_`

1. Launch SWPM with the following command:

    ```bash

>     root@sid-xxpas-0 ~]$ /usr/sap/install/SWPM/sapinst   \

Reviewer 2: I would leave the prompt part out of you example command

#### Manual DB Content Load Using Template

1. Connect to the PAS VM as `root` User
1. 1. Change to work directory and ensure the directory is empty:

>    `rm -rf /tmp/db_workdir/*; cd /tmp/db_workdir`

Reviewer: ```suggestion
   cd /tmp/db_workdir && `rm -rf *`
```

To match the wording (alternatively, change the wording).

Author: Wording changed to match command 👍

1. On the Warning Screen copy the Key ID and Key Value and store these securely and click "Ok"
1. Ensure Yes, clean up operating system users is checked
1. click "Next"
1. On the PAS nodes, a copy of the `inifile.params` file is generated in the temporary SAP installation directory:
   1. PAS inifile path `/tmp/sapinst_instdir/S4HANA2020/CORE/HDB/INSTALL/DISTRIBUTED/ABAP/APP1/inifile.params`
1. Click "Cancel" in SWPM, as the PAS installation can now be performed via the unattended method;
1. Copy and rename `inifile.params` to `pas.inifile.params` in `/tmp/app_template`:

`cp <path_to_inifile>/inifile.params /tmp/app_template/pas.inifile.params`

1. The inifiles can be used as the basis for unattended deployments
1. Create a copy of the `inifile.params` to the `sapbits` container  to the `/templates/` directory and rename to `pas.inifile.params`:

   `cp /tmp/app_template/pas.inifile.params /mnt/<sapbits fileshare path>/templates/`

`cp <path_to_inifile>/inifile.params /tmp/app_template/pas.inifile.params`

1. The inifiles can be used as the basis for unattended deployments

> 1. Create a copy of the `inifile.params` to the `sapbits` container  to the `/templates/` directory and rename to `pas.inifile.params`:

Reviewer: inifiles in `sapbits/templates` should conform to the naming convention (`<product-name>.inifile.params`).

   `cp /tmp/app_template/pas.inifile.params /mnt/<sapbits fileshare path>/templates/`

> ### Manual PAS Installation Using Template

Reviewer: These appear to duplicate the lines above.

1. Connect to PAS as `root` user
1. Change to work directory and ensure the directory is empty:
   `rm -rf /tmp/pas_workdir/*; cd /tmp/pas_workdir`
1. Launch PAS Unattended install replacing `<target vm hostname>` with the PAS VM hostname
1. For a PAS unattended install run the following:

    ```bash
    /usr/sap/install/SWPM/sapinst                                                                                         \
    SAPINST_XML_FILE=/usr/sap/install/config/MP_STACK_S4_2020_v001.xml                                                    \
    SAPINST_USE_HOSTNAME=<target vm hostname>                                                                             \
    SAPINST_EXECUTE_PRODUCT_ID=NW_ABAP_CI:S4HANA2020.CORE.HDB.ABAP                                                        \
    SAPINST_INPUT_PARAMETERS_URL=/tmp/app_template/pas.inifile.params                                                     \
    SAPINST_START_GUI=false SAPINST_START_GUISERVER=false
    ```

### Generating unattended installation parameter `inifile` for Addtional Application Server

This section covers the manual generation of the ABAP AAS (Additional Application Server) unattended install file.

> The inifile can be used for mulitple installations, therefor this process will only be required to be completed once irrespective of the number of AASes you wish to deploy.

Reviewer 2: ```suggestion
The inifile can be used for multiple installations, therefore this process will only be required to be completed once - irrespective of the number of AASes you wish to deploy.
```

:hand: To generate the AAS inifiles you must have a fully built HANA DB and ASCS.

_**Note:** Steps prefixed with * may not be encountered in 2020 versions of SAP Products._

1. Connect to the AAS VM
1. Make and change to a temporary directory:

>    1. For AAS ensure `sapinst` group exists `groupadd sapinst` then `sudo install -d -m 0777 <sid>adm -g sapinst "/tmp/aas_workdir"; cd $_`

Reviewer: ```suggestion
   1. For AAS ensure `sapinst` group exists `groupadd -g <sapinst-gid> sapinst` then `sudo install -d -m 0777 <sid>adm -g sapinst "/tmp/aas_workdir"; cd $_`
```

1. * Select No for Import ABAP Transports and click "Next"
1. * On the Preparing for the Software Update Manager Screen ensure Extract the `SUM*.SAR` Archive is checked and click "Next"
1. * On the Software Package Browser Screen check the Detected Packages table. If the Individual Package Location for SUM 2.0 is empty set the Package Path above to `/usr/sap/install/config` and click "Next"
1. * After the package location has populated, click "Next"
1. * On the Additional SAP System Languages Screen click "Next"
1. Ensure Yes, clean up operating system users is checked
1. click "Next"
1. On the AAS nodes, a copy of the `inifile.params` file is generated in the temporary SAP installation directory:
   1. AAS inifile path `/tmp/sapinst_instdir/S4HANA2020/CORE/HDB/INSTALL/AS/APPS/inifile.params`
1. Click "Cancel" in SWPM, as the AAS installation can now be performed via the unattended method;
1. Copy and rename `inifile.params` to `aas.inifile.params` in `/tmp/app_template`:

`cp <path_to_inifile>/inifile.params /tmp/app_template/aas.inifile.params`

1. The inifiles can be used as the basis for unattended deployments

> 1. Create a copy of the `inifile.params` to the `sapbits` container  to the `/templates/` directory and rename to `aas.inifile.params`:

Reviewer: Comment as above (for PAS). Files in `/sapbits/templates` should conform to the agreed naming convention.

   `cp /tmp/app_template/aas.inifile.params /mnt/<sapbits fileshare path>/templates/`

### Manual AAS Installation Using Template

:hand: A PAS must exist before the AAS Installation is attempted.

1. Connect to the AAS VM as the `root` user
1. Change to work directory and ensure the directory is empty:

>    `rm -rf /tmp/aas_workdir/*; cd /tmp/ass_workdir`

Reviewer: ```suggestion
   `rm -rf /tmp/aas_workdir/*; cd /tmp/aas_workdir`
```

🤣

   `cp /tmp/app_template/pas.inifile.params /mnt/<sapbits fileshare path>/templates/`

### Manual PAS Installation Using Template

1. Connect to PAS as `root` user
1. Change to work directory and ensure the directory is empty:
   `rm -rf /tmp/pas_workdir/*; cd /tmp/pas_workdir`
1. Launch PAS Unattended install replacing `<target vm hostname>` with the PAS VM hostname:
1. For a PAS unattended install run the following:

    ```bash
    /usr/sap/install/SWPM/sapinst                                                                                         \
    SAPINST_XML_FILE=/usr/sap/install/config/MP_STACK_S4_2020_v001.xml                                                    \
    SAPINST_USE_HOSTNAME=<target vm hostname>                                                                             \
    SAPINST_EXECUTE_PRODUCT_ID=NW_ABAP_CI:S4HANA2020.CORE.HDB.ABAP                                                        \
    SAPINST_INPUT_PARAMETERS_URL=/tmp/app_template/pas.inifile.params                                                     \
    SAPINST_START_GUI=false SAPINST_START_GUISERVER=false
    ```

> ### Generating unattended installation parameter `inifile` for Addtional Application Server

Reviewer 2: If I understand correctly, I think it's worth noting in this section that irrespective of the number of AASes you wish to deploy, this process for generating the inifile should only be required once.

This section covers the manual generation of the ABAP AAS (Additional Application Server) unattended install file.

:hand: To generate the AAS inifiles you must have a fully built HANA DB and ASCS.

_**Note:** Steps prefixed with * may not be encountered in 2020 versions of SAP Products._

1. Connect to the AAS VM
1. Make and change to a temporary directory:
   1. For AAS ensure `sapinst` group exists `groupadd sapinst` then `sudo install -d -m 0777 <sid>adm -g sapinst "/tmp/aas_workdir"; cd $_`
1. The [Access SWPM](#Access-SWPM) steps will need to be completed on the target VM before you can access SWPM

> 1. Connect to the AAS Node as Root user and launch Software Provisioning Manager, shown in [Software Provision Manager input](#Example-Software-Provision-Manager-input). Ensure that you update <sap_component> to AAS

Reviewer 2: ```suggestion
1. Connect to the AAS Node as the `root` user and launch Software Provisioning Manager, shown in [Software Provision Manager input](#Example-Software-Provision-Manager-input).
1. Ensure that you update `<sap_component>` to `AAS`
```

1. Launch the required URL to access SWPM shown in [Software Provision Manager output](#Example-Software-Provision-Manager-output)

> 1. Accept the security risk and authenticate with the systems ROOT user credentials

Reviewer 2: ```suggestion
1. Accept the security risk and authenticate with the system's `root` user credentials
```

1. Navigate through the drop-down menu:

>     1. For AAS ""SAP S/4HANA Server 2020" > "SAP HANA Database" > "Installation" > "Application Server ABAP" > "High-Availability System" > "Additional Application Server Instance"

Reviewer 2: ```suggestion
    1. For AAS, "SAP S/4HANA Server 2020" > "SAP HANA Database" > "Installation" > "Application Server ABAP" > "High-Availability System" > "Additional Application Server Instance"
```

> 1. On the Parameter Settings Screen Select "Custom" and click "Next"

Reviewer 2: ```suggestion
1. On the Parameter Settings Screen, select "Custom" and click "Next"
```

1. Ensure the Profile Directory is set to `/sapmnt/<SID>/profile/` or  `/usr/sap/<SID>/SYS/profile` and click "Next"
1. Set the Message Server Port to `36nn` where `nn` is the ASCS Instance number and click "Next"

> 1. Set the Master Password for All Users and click "Next"

Reviewer 2: Will it be obvious what to set it to?

1. * After the package location has populated, click "Next"
1. * On the Additional SAP System Languages Screen click "Next"
1. Ensure Yes, clean up operating system users is checked
1. click "Next"
1. On the AAS nodes, a copy of the `inifile.params` file is generated in the temporary SAP installation directory:
   1. AAS inifile path `/tmp/sapinst_instdir/S4HANA2020/CORE/HDB/INSTALL/AS/APPS/inifile.params`
1. Click "Cancel" in SWPM, as the AAS installation can now be performed via the unattended method;
1. Copy and rename `inifile.params` to `aas.inifile.params` in `/tmp/app_template`:

`cp <path_to_inifile>/inifile.params /tmp/app_template/aas.inifile.params`

1. The inifiles can be used as the basis for unattended deployments
1. Create a copy of the `inifile.params` to the `sapbits` container  to the `/templates/` directory and rename to `aas.inifile.params`:

   `cp /tmp/app_template/aas.inifile.params /mnt/<sapbits fileshare path>/templates/`

### Manual AAS Installation Using Template

:hand: A PAS must exist before the AAS Installation is attempted.

> 1. Connect to the AAS VM as `root` User

Reviewer 2: ```suggestion
1. Connect to the AAS VM as the `root` user
```

1. Change to work directory and ensure the directory is empty:
   `rm -rf /tmp/aas_workdir/*; cd /tmp/ass_workdir`

> 1. Launch AAS Unattended install replacing `<target vm hostname>` with the AAS VM hostname:

Reviewer 2: This steps ends with a colon, so it seems like it should be followed by an example or a command that's missing
